Poker.BE/Poker.BE.Domain/Game/Round.cs

```diff
@@ -25,69 +26,256 @@ public enum Move
+            this.currentPlayer = currentPlayer;
             this.activeUnfoldedPlayers = activeUnfoldedPlayers;
-            this.currentTurn = new Turn(currentPlayer);
+            this.currentPot = currentPot;
+            this.currentTurn = new Turn(currentPlayer, currentPot);
+            this.liveBets = new Dictionary<Player, int>();
+            foreach (Player player in activeUnfoldedPlayers)
+            {
+                LiveBets.Add(player, 0);
+            }
+            this.totalRaise = 0;
+            this.lastRaise = 0;
         }
         #endregion
 
         #region Methods
-        public void PlayMove(Move playMove)
+        public void PlayMove(Move playMove, int amountToBetOrCall)
         {
             switch (playMove)
             {
-                case Move.check :
+                case Move.check:
                     {
-                        currentTurn.Call();
+                        CurrentTurn.Check();  // Do nothing??
                         break;
                     }
-                case  Move.call:
+                case Move.call:
                     {
-                        currentTurn.Check();
+                        Call(amountToBetOrCall);
                         break;
                     }
                 case Move.fold:
                     {
-                        currentTurn.Fold();
-                        break;
+                        CurrentTurn.Fold();
+                        Player playerToRemove = this.CurrentPlayer;
+
+                        //Remover player from all pots
+                        Pot partialPotIterator = CurrentPot;
+                        while (partialPotIterator != null)
+                        {
+                            partialPotIterator.PlayersClaimPot.Remove(playerToRemove);
+                            partialPotIterator = partialPotIterator.PartialPot;
+                        }
+                        
+                        //Remove player from round
+                        this.currentPlayer = this.ActiveUnfoldedPlayers.ElementAt((ActiveUnfoldedPlayers.ToList().IndexOf(this.CurrentPlayer) - 1) % ActiveUnfoldedPlayers.Count);
+                        ActiveUnfoldedPlayers.Remove(playerToRemove);
+                        return;
+                        //break;
                     }
                 case Move.bet:
                     {
-                        currentTurn.Bet();
+                        //Check preconditions
+                        if (LastRaise > 0 || TotalRaise > 0)
+                            throw new IOException("Can't bet if someone had bet before... use raise move");
+                        if (TotalRaise + amountToBetOrCall == CurrentPlayer.Wallet.amountOfMoney + LiveBets[CurrentPlayer])
+                            throw new ArgumentOutOfRangeException("Can't bet all of your money... use all-in move");
```

Comment:
this is not `PokerException`! it will be catch only at the controller! 👎  are you sure?!?!
![image](https://cloud.githubusercontent.com/assets/9889268/26259304/145db834-3cd1-11e7-8db1-e6ba7da60e9a.png)